graspy/simulations/rdpg_corr.py

```diff
@@ -0,0 +1,114 @@
+import numpy as np
+from graspy.simulations import p_from_latent, sample_edges_corr
+
+
+def rdpg_corr(X, Y, r, rescale, directed, loops):
+    """
+    Samples a random graph based on the latent positions in X (and 
+    optionally in Y)
+    If only X :math:`\in\mathbb{R}^{n\times d}` is given, the P matrix is calculated as
+    :math:`P = XX^T`. If X, Y :math:`\in\mathbb{R}^{n\times d}` is given, then 
+    :math:`P = XY^T`. These operations correspond to the dot products between a set of 
+    latent positions, so each row in X or Y represents the latent positions in  
+    :math:`\mathbb{R}^{d}` for a single vertex in the random graph 
+    Note that this function may also rescale or clip the resulting P 
+    matrix to get probabilities between 0 and 1, or remove loops.
+    A binary random graph is then sampled from the P matrix described 
+    by X (and possibly Y).
+    Read more in the :ref:`tutorials <simulations_tutorials>`
+
+    Parameters
+    ----------
+    X: np.ndarray, shape (n_vertices, n_dimensions)
+        latent position from which to generate a P matrix
+        if Y is given, interpreted as the left latent position
+
+    Y: np.ndarray, shape (n_vertices, n_dimensions) or None, optional
+        right latent position from which to generate a P matrix
+
+    r: float
+        The value of the correlation between the same vertices in two graphs.
+
+    rescale: boolean, optional (default=True)
+        when rescale is True, will subtract the minimum value in 
+        P (if it is below 0) and divide by the maximum (if it is
+        above 1) to ensure that P has entries between 0 and 1. If
+        False, elements of P outside of [0, 1] will be clipped
+
+    directed: boolean, optional (default=False)
+        If False, output adjacency matrix will be symmetric. Otherwise, output adjacency
+        matrix will be asymmetric.
+        
+    loops: boolean, optional (default=True)
+        If False, no edges will be sampled in the diagonal. Diagonal elements in P 
+        matrix are removed prior to rescaling (see above) which may affect behavior.
+        Otherwise, edges are sampled in the diagonal.
+
+    Returns
+    -------
+    G1: ndarray (n_vertices, n_vertices)
+        A matrix representing the probabilities of connections between 
+        vertices in a random graph based on their latent positions
+        
+    G2: ndarray (n_vertices, n_vertices)
+        A matrix representing the probabilities of connections between 
+        vertices in a random graph based on their latent positions
+        
+    References
+    ----------
+    .. [1] Sussman, D.L., Tang, M., Fishkind, D.E., Priebe, C.E.  "A
```

Comment:
@SHAAAAN update this reference

Reply:
Hi Pedigo, can you tell me which reference can I refer here?

Reply:
just use this for now https://dl.acm.org/doi/10.5555/2627435.2750357

Reply:
but add the actual reference, not just the link, please